### ISSUE TITLE
SSTables: read_indexes() may return not enough entries

### ISSUE THREAD

**Issue by pdziepak** [scylladb/scylla#474]
`sstable::read_indexes()` always reads 128 entries from the index regardless of the actual index interval. The result of this is that Scylla may not see some of the partitions if the table's `min_index_interval` is high enough.

For example, if we create a table like this (using Cassandra, Scylla for now ignores `{max, min}_index_interval` setting):
```
create keyspace test with replication = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };
create table test.summary_test (a int primary key) with min_index_interval = 256; 
```
and then insert values `insert into test.summary_test (a) values (X);` for X in [1, 130], with the standard murmur partitioner the last 3 partition keys are: 77, 3, 103.
In Scylla:
`select a from test.summary_test where a = 77;` – works
`select a from test.summary_test where a = 3;` – no partition found
`select a from test.summary_test where a = 103;` – no partition found

**Comment by pdziepak** [comment]
@slivne: I can work on this.

**Comment by pdziepak** [comment]
This is fixed by 01e301fc9a65585e75e08ce9d738659c54008224.
